rotkehlchen/tests/unit/test_cost_basis.py

```diff
@@ -786,136 +823,104 @@ def test_missing_acquisitions(accountant):
+    current_total_acb += events[4].price * events[4].free_amount
+    current_amount += events[4].free_amount  # 3.5
+    assert manager.current_total_acb == current_total_acb == FVal(70)
+    assert manager.current_amount == current_amount == FVal(3.5)
+
+    # Step 6. Add a spend after a sequence of acquisitions and spends
+    add_spend(pot, amount=FVal(2), price=FVal(30))  # Sell 2 ETH for $30  pnl: 2 * (30 - 20) = $20
+    assert events[5].pnl.taxable == events[5].taxable_amount * (events[5].price - current_total_acb / current_amount) == FVal(20)  # noqa: E501
+    current_total_acb *= (current_amount - events[5].taxable_amount) / current_amount  # total acb: 70 * (3.5 - 2) / 3.5 = $30 # noqa: E501
+    current_amount -= events[5].taxable_amount  # 1.5
+    assert manager.current_total_acb == current_total_acb == FVal(30)
+    assert manager.current_amount == current_amount == FVal(1.5)
+
+    # Step 7. Check that spending up the entire remaining amount works.
+    add_spend(pot, amount=FVal(1.5), price=FVal(10))  # Sell 1.5 ETH for $10 pnl: 1.5*10 - (30/1.5)*1.5 = -$15 # noqa: E501
+    assert events[6].pnl.taxable == events[6].taxable_amount * (events[6].price - current_total_acb / current_amount) == FVal(-15)  # noqa: E501
+    current_total_acb *= (current_amount - events[6].taxable_amount) / current_amount  # total acb: 30 * (1.5 - 1.5) / 1.5 = $0  # noqa: E501
+    current_amount -= events[6].taxable_amount  # 0
+    assert manager.current_total_acb == current_total_acb == ZERO
+    assert manager.current_amount == current_amount == ZERO
+
+    # Step 8. Add one more acquisition
+    add_acquisition(pot, amount=FVal(1), price=FVal(10))  # Buy 1 ETH for $10  total acb: 0 + 1 * 10 = $10  # noqa: E501
+    assert events[7].pnl.taxable == ZERO  # No profit for acquisitions
+    current_total_acb += events[7].price * events[7].free_amount
+    current_amount += events[7].free_amount  # 1
+    assert manager.current_total_acb == current_total_acb == FVal(10)
+    assert manager.current_amount == current_amount == FVal(1)
+
+    # Step 9. Check that negative pnl is correctly handled
+    add_spend(pot, amount=FVal(0.5), price=FVal(5))  # Sell 0.5 ETH for $5  pnl: 0.5 * (5 - 10) = -$2.5  # noqa: E501
+    assert events[8].pnl.taxable == events[8].taxable_amount * (events[8].price - current_total_acb / current_amount) == FVal(-2.5)  # noqa: E501
+    current_total_acb *= (current_amount - events[8].taxable_amount) / current_amount  # total acb: 10 * (1 - 0.5) / 1 = $5  # noqa: E501
+    current_amount -= events[8].taxable_amount  # 0.5
+    assert manager.current_total_acb == current_total_acb == FVal(5)
+    assert manager.current_amount == current_amount == FVal(0.5)
+
+    # Step 10. Try to spend more than the remaining amount
+    add_spend(pot, amount=FVal(0.6), price=FVal(5))  # Sell 0.6 ETH for $5  pnl: 0.5 * (5 - 10) + (0.1 * 5 <-- amount that has no matching acquisition) = -$2  # noqa: E501
```

Comment:
This needs better explanation

```suggestion
    add_spend(pot, amount=FVal(0.6), price=FVal(5))  # Sell 0.6 ETH for $5
    # pnl: pnl_for_known_acquisition + pnl_for_rest_uses_full_price_as_profit
    # 0.5*5 - (5/0.5)*0.5 + 0.1 * 5 = -2$
```